src/api/hooks/useGetANS.ts

```diff
@@ -2,25 +2,40 @@ import {useEffect, useState} from "react";
 import {NetworkName} from "../../constants";
 import {useGlobalState} from "../../GlobalState";
 
-function getFetchNameUrl(network: NetworkName, address: string) {
+function getFetchNameUrl(
+  network: NetworkName,
+  address: string,
+  isPrimary: boolean,
+) {
   if (network !== "testnet" && network !== "mainnet") {
     return undefined;
   }
 
-  return `https://www.aptosnames.com/api/${network}/v1/name/${address}`;
+  return isPrimary
+    ? `https://www.aptosnames.com/api/${network}/v1/primary-name/${address}`
+    : `https://www.aptosnames.com/api/${network}/v1/name/${address}`;
 }
 
 export function useGetNameFromAddress(address: string) {
   const [state, _] = useGlobalState();
   const [name, setName] = useState<string | undefined>();
-  const url = getFetchNameUrl(state.network_name, address);
 
   useEffect(() => {
-    if (url !== undefined) {
+    const primaryNameUrl = getFetchNameUrl(state.network_name, address, true);
+    if (primaryNameUrl !== undefined) {
       const fetchData = async () => {
-        const response = await fetch(url);
-        const {name} = await response.json();
```

Comment:
saw there are three places(here, line 75-76, line 27-28) doing the same thing: fetch url and then get json str from url response. would suggest we consolidate all three of them by having a helper function perform this logic.